gnomad_qc/v5/sample_qc/relatedness.py

```diff
@@ -0,0 +1,180 @@
+) -> None:
+    """
+    Print the command to submit a dsub job for running cuKING.
+
+    :param cuking_input_path: Path to the cuKING input Parquet files.
+    :param cuking_output_path: Path to the cuKING output Parquet files.
+    :param min_emission_kinship: Minimum kinship threshold for emitting a pair of
+        samples in the relatedness output.
+    :param cuking_split_factor: Split factor to use for splitting the full relatedness
+        matrix table into equally sized submatrices that are computed independently to
+        parallelize the relatedness computation and decrease the memory requirements.
+        For example, to halve memory requirements, the full matrix can be split into
+        equally sized submatrices (i.e. a 'split factor' of 4). Only the
+        'upper triangular' submatrices need to be evaluated due to the symmetry of the
+        relatedness matrix, leading to 10 shards. Default is 4.
+    :param location: Location to run the dsub job. Default is "us-central1".
+    :param machine_type: Machine type to use for the dsub job. Default is "a2-highgpu-1g".
+    :param accelerator_count: Number of accelerators to use for the dsub job. Default is 1.
+    :param accelerator_type: Type of accelerator to use for the dsub job.
+        Default is "nvidia-tesla-a100".
+    :param requester_pays_project: Project ID to use for requester pays buckets.
+        Default is "terra-vpc-sc-93ccd8d2".
+    :return:
+    """
+    logger.info(
+        "Printing a command that can be used to submit a dsub job to run "
+        "cuKING on the files created by --prepare-cuking-inputs."
+    )
+    logger.warning(
+        "This printed command assumes that the cuKING directory is in the same "
+        "location where the command is being run and that $PROJECT_ID is set!"
+    )
+    SPLIT_FACTOR = cuking_split_factor
+    TOTAL_SHARDS = SPLIT_FACTOR * (SPLIT_FACTOR + 1) / 2
+    print(
+        textwrap.dedent(
+            f"""\
+            for SHARD_INDEX in $(seq 0 $(({TOTAL_SHARDS} - 1))); do \\
+                cuKING_dsub \\
+                --location={location} \\
+                --machine-type={machine_type} \\
+                --accelerator-count={accelerator_count} \\
+                --accelerator-type={accelerator_type} \\
+                --command="cuking \\
+                --input_uri={cuking_input_path} \\
+                --output_uri={cuking_output_path} \\
+                --requester_pays_project={requester_pays_project} \\
+                --kin_threshold={min_emission_kinship} \\
+                --split-factor={SPLIT_FACTOR} \\
+                --shard-index=SHARD_INDEX" \\
+            done
+            """
+        )
+    )
+
+
+def main(args):
+    """Compute relatedness estimates among pairs of samples in the callset."""
+    # test = args.test
+    # overwrite = args.overwrite
+    min_emission_kinship = args.min_emission_kinship
+
+    if args.print_cuking_command:
```

Comment:
Where is the QC MT converted to the parquet files?

Reply:
this will be done in a notebook (https://workbench.researchallofus.org/workspaces/aou-rw-4b0bd63f/gnomadproduction/analysis/preview/relatedness_02_prep_AoU_and_gnomAD_data_for_cuKING.ipynb) -- I initially had the code in this PR and ran into an import error. I also noticed that Mike or Kristen commented in another notebook that the Hail init needs to include a tmp dir to avoid memory issues so was planning to run this conversion in the notebook above following the example set on v7 (see comment for more details)
![image](https://github.com/user-attachments/assets/f16fdfb2-d47e-416e-9efd-4a781305f494)

Reply:
also -- happy to put the code back here if you happen to know how to fix the import error 😆